### ISSUE TITLE
bosh deploy fails while updating job runner_z1 > runner_z1/0

### ISSUE THREAD

**Issue by MoizArif** [cloudfoundry/cf-release#1021]
Hi all,

i am getting this error which deploying cf.

```
  Started updating job runner_z1 > runner_z1/0 (13743010-281b-4601-b117-479472016342) (canary). Failed: 'runner_z1/0 (13743010-281b-4601-b117-479472016342)' is not running after update. Review logs for failed jobs: warden, dea_next (00:11:00)

Error 400007: 'runner_z1/0 (13743010-281b-4601-b117-479472016342)' is not running after update. Review logs for failed jobs: warden, dea_next
```

All processes on the runner_z1/0 vm are running:
```
root@79c4cc9f-6b6d-49d4-819d-77048cbfd1a1:/home/vcap# monit summary
The Monit daemon 5.2.5 uptime: 20m 

Process 'consul_agent'              running
Process 'warden'                    running
Process 'dea_next'                  running
Process 'dir_server'                running
Process 'dea_logging_agent'         running
Process 'metron_agent'              running
System 'system_localhost'           running
```

The warden error logs report, /var/vcap/sys/log/warden_ctl.err.log:
```
[2016-08-01 08:23:55+0000] ------------ STARTING warden_ctl at Mon Aug  1 08:23:54 UTC 2016 --------------
[2016-08-01 08:26:42+0000] ------------ STARTING warden_ctl at Mon Aug  1 08:26:42 UTC 2016 --------------
[2016-08-01 08:28:54+0000] ------------ STARTING warden_ctl at Mon Aug  1 08:28:54 UTC 2016 --------------
[2016-08-01 08:31:06+0000] ------------ STARTING warden_ctl at Mon Aug  1 08:31:06 UTC 2016 --------------
[2016-08-01 08:33:20+0000] ------------ STARTING warden_ctl at Mon Aug  1 08:33:20 UTC 2016 --------------
[2016-08-01 08:35:34+0000] ------------ STARTING warden_ctl at Mon Aug  1 08:35:34 UTC 2016 --------------
[2016-08-01 08:37:54+0000] ------------ STARTING warden_ctl at Mon Aug  1 08:37:54 UTC 2016 --------------
[2016-08-01 08:38:54+0000] rake aborted!
[2016-08-01 08:38:54+0000] Warden::WardenError: command exited with failure
[2016-08-01 08:38:54+0000] Tasks: TOP => warden:start
[2016-08-01 08:38:54+0000] (See full trace by running task with --trace)
[2016-08-01 08:38:54+0000] rake aborted!
[2016-08-01 08:38:54+0000] rake aborted!
[2016-08-01 08:38:54+0000] rake aborted!
[2016-08-01 08:38:54+0000] rake aborted!
[2016-08-01 08:38:54+0000] no acceptor (port is in use or requires root privileges)
[2016-08-01 08:38:54+0000] no acceptor (port is in use or requires root privileges)
[2016-08-01 08:38:54+0000] no unix-domain acceptor
[2016-08-01 08:38:54+0000] Tasks: TOP => warden:start
[2016-08-01 08:38:54+0000] no acceptor (port is in use or requires root privileges)
[2016-08-01 08:38:54+0000] Tasks: TOP => warden:start
[2016-08-01 08:38:54+0000] Tasks: TOP => warden:start
[2016-08-01 08:38:54+0000] (See full trace by running task with --trace)
[2016-08-01 08:38:54+0000] Tasks: TOP => warden:start
[2016-08-01 08:38:54+0000] (See full trace by running task with --trace)
[2016-08-01 08:38:54+0000] (See full trace by running task with --trace)
[2016-08-01 08:38:54+0000] (See full trace by running task with --trace)
[2016-08-01 08:39:01+0000] rake aborted!
[2016-08-01 08:39:01+0000] no acceptor (port is in use or requires root privileges)
[2016-08-01 08:39:01+0000] Tasks: TOP => warden:start
[2016-08-01 08:39:01+0000] (See full trace by running task with --trace)
[2016-08-01 08:39:52+0000] rake aborted!
[2016-08-01 08:39:52+0000] SignalException: SIGTERM
[2016-08-01 08:39:52+0000] Tasks: TOP => warden:start
[2016-08-01 08:39:52+0000] (See full trace by running task with --trace)
```

/var/vcap/sys/log/dea_ctl.err.log reports:
```
[2016-08-01 08:26:33+0000] ------------ STARTING dea_ctl at Mon Aug  1 08:26:33 UTC 2016 --------------
[2016-08-01 08:28:42+0000] ------------ STARTING dea_ctl at Mon Aug  1 08:28:42 UTC 2016 --------------
[2016-08-01 08:30:54+0000] ------------ STARTING dea_ctl at Mon Aug  1 08:30:54 UTC 2016 --------------
[2016-08-01 08:33:06+0000] ------------ STARTING dea_ctl at Mon Aug  1 08:33:06 UTC 2016 --------------
[2016-08-01 08:35:20+0000] ------------ STARTING dea_ctl at Mon Aug  1 08:35:20 UTC 2016 --------------
[2016-08-01 08:37:34+0000] ------------ STARTING dea_ctl at Mon Aug  1 08:37:34 UTC 2016 --------------
[2016-08-01 08:39:00+0000] ------------ STARTING dea_ctl at Mon Aug  1 08:39:00 UTC 2016 --------------
[2016-08-01 08:39:54+0000] ------------ STARTING dea_ctl at Mon Aug  1 08:39:54 UTC 2016 --------------

```

/var/vcap/sys/log/dea_ctl.log reports:
```
[2016-08-01 08:26:33+0000] dea.service.cf.internal has address 10.10.0.107
[2016-08-01 08:26:34+0000] ------------ STARTING dea_ctl at Mon Aug  1 08:26:33 UTC 2016 --------------
[2016-08-01 08:26:41+0000] Killing /var/vcap/sys/run/dea_next/dea_next.pid: 4657
[2016-08-01 08:26:41+0000] .Stopped
[2016-08-01 08:28:42+0000] dea.service.cf.internal has address 10.10.0.107
[2016-08-01 08:28:42+0000] ------------ STARTING dea_ctl at Mon Aug  1 08:28:42 UTC 2016 --------------
[2016-08-01 08:28:53+0000] Killing /var/vcap/sys/run/dea_next/dea_next.pid: 5192
[2016-08-01 08:28:54+0000] .Stopped
[2016-08-01 08:30:54+0000] dea.service.cf.internal has address 10.10.0.107
[2016-08-01 08:30:54+0000] ------------ STARTING dea_ctl at Mon Aug  1 08:30:54 UTC 2016 --------------
[2016-08-01 08:31:05+0000] Killing /var/vcap/sys/run/dea_next/dea_next.pid: 5462
[2016-08-01 08:31:06+0000] .Stopped
[2016-08-01 08:33:06+0000] dea.service.cf.internal has address 10.10.0.107
[2016-08-01 08:33:06+0000] ------------ STARTING dea_ctl at Mon Aug  1 08:33:06 UTC 2016 --------------
[2016-08-01 08:33:19+0000] Killing /var/vcap/sys/run/dea_next/dea_next.pid: 5721
[2016-08-01 08:33:20+0000] ..Stopped
[2016-08-01 08:35:20+0000] dea.service.cf.internal has address 10.10.0.107
[2016-08-01 08:35:20+0000] ------------ STARTING dea_ctl at Mon Aug  1 08:35:20 UTC 2016 --------------
[2016-08-01 08:35:31+0000] Killing /var/vcap/sys/run/dea_next/dea_next.pid: 5989
[2016-08-01 08:35:33+0000] ...Stopped
[2016-08-01 08:37:34+0000] dea.service.cf.internal has address 10.10.0.107
[2016-08-01 08:37:34+0000] ------------ STARTING dea_ctl at Mon Aug  1 08:37:34 UTC 2016 --------------
[2016-08-01 08:37:50+0000] Killing /var/vcap/sys/run/dea_next/dea_next.pid: 6257
[2016-08-01 08:37:52+0000] ...Stopped
[2016-08-01 08:39:00+0000] dea.service.cf.internal has address 10.10.0.107
[2016-08-01 08:39:01+0000] ------------ STARTING dea_ctl at Mon Aug  1 08:39:00 UTC 2016 --------------
[2016-08-01 08:39:51+0000] Killing /var/vcap/sys/run/dea_next/dea_next.pid: 6582
[2016-08-01 08:39:52+0000] .Stopped
[2016-08-01 08:39:54+0000] dea.service.cf.internal has address 10.10.0.107
[2016-08-01 08:39:54+0000] ------------ STARTING dea_ctl at Mon Aug  1 08:39:54 UTC 2016 --------------
```

What seems to the issue with dea & warden?

Here is the output of bosh vms
```
root@node-42:~/cf-release# bosh vms
RSA 1024 bit CA certificates are loaded due to old openssl compatibility
Acting as user 'admin' on 'my-bosh'
Deployment 'cf'

Director task 136

Task 136 done

+---------------------------------------------------------------------------+---------+-----+-----------+---------------+
| VM                                                                        | State   | AZ  | VM Type   | IPs           |
+---------------------------------------------------------------------------+---------+-----+-----------+---------------+
| api_worker_z1/0 (0488a859-27a9-4caf-96d2-7da70ab9a52e)                    | running | n/a | small_z1  | 10.10.0.105   |
| api_z1/0 (2d457b4d-b352-44c7-b1b0-960d0839c82f)                           | running | n/a | large_z1  | 10.10.0.104   |
| blobstore_z1/0 (e66a50c2-4349-4ab7-a022-66624b12769b)                     | running | n/a | medium_z1 | 10.10.0.102   |
| consul_z1/0 (6530523f-936f-4cc3-a7f3-03640705a70b)                        | running | n/a | small_z1  | 10.10.0.137   |
| doppler_z1/0 (a71d1dae-fc8b-4038-823f-003f515fc286)                       | running | n/a | medium_z1 | 10.10.0.108   |
| etcd_z1/0 (058b2b97-add3-4901-bda3-b080120c07db)                          | running | n/a | medium_z1 | 10.10.0.133   |
| ha_proxy_z1/0 (fb18fe8b-efa8-4243-8cc6-c0ef3f70e809)                      | running | n/a | router_z1 | 192.168.14.90 |
|                                                                           |         |     |           | 10.10.0.125   |
| hm9000_z1/0 (36a3c71f-b7a1-4107-ba3e-3b62defd2557)                        | running | n/a | medium_z1 | 10.10.0.106   |
| loggregator_trafficcontroller_z1/0 (56f0b8f2-dc7e-4f81-a05f-271817297f29) | running | n/a | small_z1  | 10.10.0.109   |
| nats_z1/0 (2ba05347-ed29-4ee6-bdda-66c085ece8b5)                          | running | n/a | medium_z1 | 10.10.0.127   |
| postgres_z1/0 (b66b97a4-96c6-4cd8-a722-9e2a63b1a4bb)                      | running | n/a | medium_z1 | 10.10.0.129   |
| router_z1/0 (8b3debc1-894d-40d5-b270-3c8b56075f17)                        | running | n/a | router_z1 | 10.10.0.130   |
| runner_z1/0 (4aa93ab1-58eb-4bb3-899c-b010ce6744a7)                        | running | n/a | runner_z1 | 10.10.0.107   |
| stats_z1/0 (9780729c-eb40-41d8-b3dd-62a23f04b538)                         | running | n/a | small_z1  | 10.10.0.101   |
| uaa_z1/0 (3562f654-981c-4074-961c-0a24c10ad530)                           | running | n/a | medium_z1 | 10.10.0.103   |
+---------------------------------------------------------------------------+---------+-----+-----------+---------------+

VMs total: 15
```

Here is my cf-deployment.yml file: http://pastebin.com/Dc3Geuzc

**Comment by dsabeti** [comment]
Hi @MoizArif 

It looks like your warden server is having trouble coming up because it's  trying to use a port that's already in use:
```
...
[2016-08-01 08:39:01+0000] rake aborted!
[2016-08-01 08:39:01+0000] no acceptor (port is in use or requires root privileges)
```

As far as we can tell, the port is hard-coded to `2345`, so it seems unlikely that you've misconfigured anything. It might be that you have another process that has already claimed that port. Can you post the output of `lsof -i` on the `runner_z1/0` vm?

If the port isn't already in use, it could be that you're asking for a port that requires root privileges, but the process isn't being started by a root user.

Also, we're going to rope in the team that works on the DEA and Warden. @fraenkel and @swetharepakula, do y'all have any thoughts about how to debug this?

Thanks,
David && @jaresty 
CF Release Integration Team

**Comment by MoizArif** [comment]
Thanks @dsabeti for the reply. Yes, i figured the same. 

I was able to solve the issue. I made some changes to my CF deployment manifest file and it got fixed. The changes i made were related to ssl which i disabled just for testing.

On the deployed CF setup here is the output of the command:
```
root@f7eb8006-86a6-400b-9768-708f516b721d:/home/vcap# lsof -i | grep 2345
ruby       9285   root   15u  IPv4   25817      0t0  TCP localhost:2345 (LISTEN)
ruby       9428   root   15u  IPv4   25039      0t0  TCP localhost:23456 (LISTEN)
```

which seems to be correct. 

At the moment i can create applications and access them too.
